src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ILScanner.cs

```diff
@@ -437,5 +442,43 @@ public override bool IsEffectivelySealed(MethodDesc method)
                 return base.IsEffectivelySealed(method);
             }
         }
+
+        private class ScannedInliningPolicy : IInliningPolicy
+        {
+            private readonly HashSet<TypeDesc> _constructedTypes = new HashSet<TypeDesc>();
+            private readonly CompilationModuleGroup _baseGroup;
+
+            public ScannedInliningPolicy(CompilationModuleGroup baseGroup, ImmutableArray<DependencyNodeCore<NodeFactory>> markedNodes)
+            {
+                _baseGroup = baseGroup;
+
+                foreach (var node in markedNodes)
+                {
+                    if (node is ConstructedEETypeNode eetypeNode)
+                    {
+                        TypeDesc type = eetypeNode.Type;
+                        _constructedTypes.Add(type);
+                    }
+                }
+            }
+
+            public bool CanInline(MethodDesc caller, MethodDesc callee)
+            {
+                if (_baseGroup.CanInline(caller, callee))
+                {
+                    /// Since the scanner doesn't look at instance methods whose owning type
```

Comment:
This is not a doc block. Use double-slash instead of triple-slash

Reply:
I never understood how these doc blocks within a method work - do they do anything.

They enable VS to understand the `<see cref=...>`. Otherwise it's just a text string that is going to start bitrotting the moment it's checked in.

Reply:
C# compiler warning "CS1587: XML comment is not placed on a valid language element" suggests that this is not meant to work. We do not have this warning enabled for the compiler, but it is enabled in most of the tree.